precompiles/xcm/src/lib.rs

```diff
@@ -48,30 +55,66 @@ mod tests;
 pub enum Action {
     AssetsWithdrawNative = "assets_withdraw(address[],uint256[],bytes32,bool,uint256,uint256)",
     AssetsWithdrawEvm = "assets_withdraw(address[],uint256[],address,bool,uint256,uint256)",
-    RemoteTransact = "remote_transact(uint256,bool,address,uint256,bytes,uint64)",
+    RemoteTransactOld = "remote_transact(uint256,bool,address,uint256,bytes,uint64)",
     AssetsReserveTransferNative =
         "assets_reserve_transfer(address[],uint256[],bytes32,bool,uint256,uint256)",
     AssetsReserveTransferEvm =
         "assets_reserve_transfer(address[],uint256[],address,bool,uint256,uint256)",
+    AssetsWithdraw = "assets_withdraw(address[],uint256[],(uint8,bytes[]),(uint8,bytes[]),uint256)",
+    RemoteTransactNew = "remote_transact((uint8,bytes[]),address,uint256,bytes,uint64)",
+    AssetsReserveTransfer =
+        "assets_reserve_transfer(address[],uint256[],(uint8,bytes[]),(uint8,bytes[]),uint256)",
+    SendXCM = "send_xcm((uint8,bytes[]),bytes)",
+    XtokensTransfer = "transfer(address,uint256,(uint8,bytes[]),uint64)",
```

Comment:
Most of the function interfaces are wrong here. It should not end by uint64 but (uint64, uint64) as you introduced new Weight struct. So all xtokens calls panics in `let selector = handle.read_selector()?;`